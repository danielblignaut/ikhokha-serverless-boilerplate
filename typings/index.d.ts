declare module 'serverless-offline'
declare module 'serverless-webpack'
declare module 'custom-env' 

declare module '*.json' {
	const content: any
	export default content
}

declare module '*.graphql' {
	const content: any
	export default content
}

declare namespace NodeJS {
	interface ProcessEnv {
		PROXY_API_URL: string
		PROXY_API_TOKEN: string
		PROXY_API_TOKEN_KEY: string
		TEST__SERVER_URL: string
		NODE_ENV: 'production' | 'test' | 'development'
		WEBPACK_SERVE: boolean
	}
}