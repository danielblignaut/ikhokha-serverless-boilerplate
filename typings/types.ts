import { ApolloLink } from 'apollo-link'
import { GraphQLSchema } from 'graphql'

export interface CustomContext {

}

export interface OperationsToKeep {
	Query: string[]
	Mutation: string[]
	[key: string]: string[]
}

export interface AwsServer {
	
}

export interface ProxyLinkHeaders {
	[key: string]: string
}

export interface ProxyLinkContext {
	headers: ProxyLinkHeaders
}


export interface TypesToKeep extends Array<string> {

}

export interface FsMock {
	__setMockFiles: (files: string[])=> void
	existsSync: (path: string)=> boolean
	ReadStream: Function
	WriteStream: Function
	promises: {
		writeFile: (path: string, content: string)=> Promise<void>
	}
}

export interface GraphQlToolsMock {
	introspectSchema: (link: ApolloLink)=> GraphQLSchema
}

export interface GraphQlMockError {

}

export interface GraphQlMockResult {
	body: string
}

export interface GraphQlMockResponseBody<T = any> {
	data: T
}