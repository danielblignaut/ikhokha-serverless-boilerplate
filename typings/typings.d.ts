declare module '*.graphql' {
	const content: any
	export default content
}

declare module '*.json' {
	const content: any
	export default content
}

declare module 'serverless-webpack' {
	const lib: {
		webpack: {
			isLocal: boolean
		}
		
		entries: any
	}
	export { lib }
}
