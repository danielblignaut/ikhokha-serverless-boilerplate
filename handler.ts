import createServer from '@ikhokha/src/create-server'
import { APIGatewayProxyEvent, Callback, APIGatewayProxyResult, Context } from 'aws-lambda'
import {CustomContext} from '@ikhokha/typings/types'


const context: CustomContext = {

}

export const appFunction: (event: APIGatewayProxyEvent, context: Context, callback: Callback<APIGatewayProxyResult>)=> void = createServer(context)
