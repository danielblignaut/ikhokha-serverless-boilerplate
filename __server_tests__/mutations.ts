import {appFunction} from '../handler'
import { APIGatewayEvent, APIGatewayProxyResult } from 'aws-lambda'
import { Context } from 'aws-lambda'
import createLambdaContext from '@ikhokha/test/create-lambda-context'
import createLambdaEvent from '@ikhokha/test/create-lambda-event'
import * as queries from '@ikhokha/test/graphql-files/queries'
import * as mutations from '@ikhokha/test/graphql-files/mutations'
import { GraphQlMockError, GraphQlMockResult, GraphQlMockResponseBody } from '@ikhokha/typings/types'


describe('test Shopify Queries', () => {
	let event: APIGatewayEvent
	let context: Context
	let callback: Function

	test('create a checkout', (done)=>{

		const callback = (error: string | Error | null | undefined, result: APIGatewayProxyResult | undefined) => {
			if(typeof result === 'undefined') return
			
			const body = JSON.parse(result.body) as GraphQlMockResponseBody
			expect(body.data).toMatchSnapshot({
				checkoutCreate: {
					checkout: {
						id: expect.any(String)
					}
				}
			})
			done()
		}
		event = createLambdaEvent('POST',mutations.createCheckout,{
			email: 'daniel@ucook.co.za',
			note: 'this is my test note!!!'
		})
		context = createLambdaContext({}, callback)
		const app = appFunction(event, context, callback)
	})
})