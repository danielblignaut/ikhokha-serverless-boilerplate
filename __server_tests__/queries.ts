import {appFunction} from '../handler'
import { APIGatewayEvent, APIGatewayProxyResult } from 'aws-lambda'
import { Context } from 'aws-lambda'
import createLambdaContext from '@ikhokha/test/create-lambda-context'
import createLambdaEvent from '@ikhokha/test/create-lambda-event'
import * as queries from '@ikhokha/test/graphql-files/queries'
import * as mutations from '@ikhokha/test/graphql-files/mutations'
import { GraphQlMockError, GraphQlMockResult, GraphQlMockResponseBody } from '@ikhokha/typings/types'


describe('test Shopify Queries', () => {
	let event: APIGatewayEvent
	let context: Context
	let callback: Function

	test('succesfully get first product', (done)=>{

		const callback = (error: string | Error | null | undefined, result: APIGatewayProxyResult | undefined) => {
			if(typeof result === 'undefined') return
			
			const body = JSON.parse(result.body) as GraphQlMockResponseBody
			expect(body.data).toMatchSnapshot()
			done()
		}
		
		event = createLambdaEvent('GET',queries.firstNProducts,{
			first: 1
		})
		context = createLambdaContext({}, callback)
		const app = appFunction(event, context, callback)
	})

})