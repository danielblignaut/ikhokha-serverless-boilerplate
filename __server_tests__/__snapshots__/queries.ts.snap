// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Shopify Queries succesfully get first product 1`] = `
Object {
  "products": Object {
    "edges": Array [
      Object {
        "node": Object {
          "id": "Z2lkOi8vc2hvcGlmeS9Qcm9kdWN0LzE2ODA1ODQ2Mzg1MjI=",
          "title": "Mover Lite",
        },
      },
    ],
  },
}
`;
